File: baekjoon/two-pointer/python/BOJ1806.py
from sys import stdin
readline = stdin.readline

N, S = map(int, readline().split())
arr = list(map(int, readline().split()))
INF = float('inf')

def solve():
    left, curr, ans = 0, 0, INF

    for right in range(N):
        curr += arr[right]

        while left < right and curr - arr[left] >= S:
            curr -= arr[left]
            left += 1
        
        if curr >= S and ans > right - left:
            ans = right - left
        
    print(ans + 1 if ans != INF else 0)

solve() 

File: 0_1.baekjoon/two-pointer/java/BOJ1806.java
import java.io.*;
import java.util.*;

class BOJ1806 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int N = Integer.parseInt(st.nextToken());
        int S = Integer.parseInt(st.nextToken());

        st = new StringTokenizer(br.readLine());
        int[] arr = new int[N];
        for (int i = 0; i < N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
        }

        int left = 0, right = 0, curr = 0;
        int ans = Integer.MAX_VALUE;

        while (right < N) {
            curr += arr[right];

            if (curr >= S) {

                while (left < right && curr - arr[left] >= S) {
                    curr -= arr[left++];
                }

                if (ans > right - left + 1) {
                    ans = right - left + 1;
                }
            }

            right++;
        }
        System.out.println(ans == Integer.MAX_VALUE ? 0 : ans);
    }
}